cmake/Provenance.cmake

```diff
@@ -0,0 +1,119 @@
+#========================================================================================
+# Parthenon performance portable AMR framework
+# Copyright(C) 2020-2025 The Parthenon collaboration
+# Licensed under the 3-clause BSD License, see LICENSE file for details
+#========================================================================================
+# (C) (or copyright) 2020-2025. Triad National Security, LLC. All rights reserved.
+#
+# This program was produced under U.S. Government contract 89233218CNA000001 for Los
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#========================================================================================
+# Originally based on code by Jonathan Hamberg
+# https://gitlab.com/jhamberg/cmake-examples/-/tree/master/cmake
+set(CURRENT_LIST_DIR ${CMAKE_CURRENT_LIST_DIR})
+if (NOT DEFINED pre_configure_dir)
+    set(pre_configure_dir ${CMAKE_SOURCE_DIR}/src)
+endif ()
+
+if (NOT DEFINED post_configure_dir)
+    set(post_configure_dir ${CMAKE_SOURCE_DIR}/src/generated)
```

Comment:
Should we be generating this in `CMAKE_SOURCE_DIR` or in `CMAKE_BINARY_DIR`? The binary dir seems cleaner if possible so that we aren't putting untracked files in the source tree. I guess another option would be to add `src/generate/*` to `.gitignore`

Reply:
:+1: I agree. I think it should go in `CMAKE_BINARY_DIR` but `.gitignore` would be an acceptable solution too

Reply:
Generated files now live in `$CMAKE_BINARY_DIR/generated`